blob/local_store.go

```diff
@@ -3,26 +3,42 @@ package blob
 import (
 	"context"
 	"errors"
+	"fmt"
 	"io"
 	"os"
 	"path/filepath"
+
+	"github.com/gammazero/fsutil/disk"
 )
 
 var _ Store = (*LocalStore)(nil)
 
+const (
+	Kib = 1 << (10 * (iota + 1))
+	Mib
+	Gib
+)
+
+const defaultMinFreeSpace = 64 * Mib
+
 // LocalStore is a Store that stores blobs as flat files in a configured directory.
 // Blobs are stored as flat files, named by their ID with .bin extension.
 // This store is used primarily for testing purposes.
 type LocalStore struct {
-	dir string
+	dir          string
+	minFreeSpace uint64
 }
 
 // NewLocalStore instantiates a new LocalStore and uses the given dir as the place to store blobs.
 // Blobs are stored as flat files, named by their ID with .bin extension.
-func NewLocalStore(dir string) *LocalStore {
+func NewLocalStore(dir string, minFreeSpace uint64) *LocalStore {
```

Comment:
This new parameter might go better in a new config struct, since it's optional and what it represents is not obvious when reading a function call. Will also set up a better pattern for adding new options if ever needed in the future.

`myStore := NewLocalStore("/a/dir", 1024) // what is 1024?`

`myStore := NewLocalStore("/a/dir", LocalStoreConfig { MinFreeSpace: 1024 }) // better in my opinion`

Reply:
I would prefer to pass Option parameters instead of a config. If passing a config, we would still want a source of default values, which the option provides.  I am implemented the optional parameter in the PR update.

As far as this: `myStore := NewLocalStore("/a/dir", 1024)`
... no one should ever do that anyway.  Code like that should always use a `const`, never a raw number. For example:
```go
const minFreeSpace = 2 * Gib // 2 Gib must remain free on disk
...
myStore := NewLocalStore("/a/dir", minFreeSpace)
```

Reply:
nice, option pattern is even better

fair point about minFreeSpace as a param. i guess i can't back it up more than that it feels weird - i like to see fundamental, required fields as direct constructor params and optional peripheral fields to go in a config struct or option pattern. 

definitely more philosophical than practical, would never deny a PR on these grounds, i just like discussing style : )